mtp_api/apps/mtp_auth/views.py

```diff
@@ -35,19 +47,31 @@ class UserViewSet(viewsets.ModelViewSet):
     serializer_class = UserSerializer
 
     def get_queryset(self):
-        client_id = self.request.auth.application.client_id
-        queryset = User.objects.filter(
-            applicationusermapping__application__client_id=client_id,
-        ).order_by('username')
-
-        user_prisons = PrisonUserMapping.objects.get_prison_set_for_user(self.request.user)
-        if len(user_prisons) > 0:
-            prison_filters = []
-            for prison in user_prisons:
-                prison_filters.append(Q(prisonusermapping__prisons=prison))
-            queryset = queryset.filter(
-                reduce(lambda a, b: a | b, prison_filters)).distinct()
-        return queryset
+        """
+        Set of users for user account management AND looking up details of self.
+        Set is determined by matching set of prisons if some exist OR by key group as determined by Role models.
+        If a user falls into multiple roles, they can only edit themselves.
+        """
+        user = self.request.user
+        queryset = User.objects.filter(is_superuser=user.is_superuser).order_by('username')
+
+        try:
+            prisons = list(PrisonUserMapping.objects.get(user=user).prisons.values_list('pk', flat=True))
+        except PrisonUserMapping.DoesNotExist:
+            prisons = []
```

Comment:
Could use `PrisonUserMapping.objects.get_prison_set_for_user` to avoid having to handle `DoesNotExist` here.